packages/scene-composer/src/components/panels/scene-settings/MatterportIntegration.tsx

```diff
@@ -64,15 +66,17 @@ export const MatterportIntegration: React.FC = () => {
     return () => unsubscribe(onUpdated);
   }, []);
 
-  const onConnectionNameChange = useCallback(async (event) => {
-    const newConnection = event.detail.selectedOption.value;
-    setSelectedConnectionName(newConnection);
-
-    if (twinMakerSceneMetadataModule) {
-      const updatedSceneInfo = await getUpdatedSceneInfoForConnection(newConnection, twinMakerSceneMetadataModule);
-      twinMakerSceneMetadataModule.updateSceneInfo(updatedSceneInfo);
-    }
-  }, []);
+  const onConnectionNameChange = useCallback(
+    async (event) => {
+      const newConnection = event.detail.selectedOption.value;
+      if (twinMakerSceneMetadataModule) {
+        const updatedSceneInfo = await getUpdatedSceneInfoForConnection(newConnection, twinMakerSceneMetadataModule);
+        await twinMakerSceneMetadataModule.updateSceneInfo(updatedSceneInfo);
+      }
+      setSelectedConnectionName(newConnection);
```

Comment:
will the UI be delayed to refresh with the new selection if you set state after the update API?

Reply:
No delay observed. Expected to save first before updating the state variable as it will trigger the loading of viewer